src/modals/WalletConnectModal.tsx

```diff
@@ -41,22 +43,29 @@ enum State {
 }
 
 const WalletConnectModal = ({ onClose, onConnect }: Props) => {
-  const { addresses } = useAddressesContext()
+  const { addresses, mainAddress } = useAddressesContext()
   const { walletConnect } = useWalletConnectContext()
   const [uri, setUri] = useState('')
   const [state, setState] = useState(addresses.length > 0 ? State.InitiateSession : State.RequireUnlock)
   const [proposal, setProposal] = useState<SessionTypes.Proposal>()
-  const [permittedChain, setPermittedChain] = useState('')
+  const [permittedChain, setPermittedChain] = useState({
+    chainId: formatChain(0, -1),
+    networkId: 0,
+    permittedGroup: -1
```

Comment:
Good question. It could be `-1, 0, 1, 2, 3`. Positive number means a specific group, while `-1` means all groups are allowed.

I am not super happy about using `-1` for all groups. Looking for better alternatives

Reply:
Why not when the group is not specified it's all groups?

What does `-2` then mean?

Reply:
type Group = number | undefined

Reply:
And when `group` is an argument in a function: `function a(group?: number, arg2, arg3) { ... }`

Reply:
`-2` will be treated as an unknown group, and will not be compatible with any group: [code](https://github.com/alephium/walletconnect/pull/1/files#diff-a2a171449d862fe29692ce031981047d7ab755ae7f84c707aef80701b3ea0c80R316-R318)

Protocol level: I think using `-1` at the protocol level is lightweight and efficient, but I should hide these technical details from developers. 
API level: `number | undefined` sounds great to go. I will play with this idea and see if there is anything I missed.

Reply:
The only issue with `-1` at the protocol level is a typo: if someone meant `1` but typos `-1` (for whatever reason, maybe they are tired), it will have very, very different meaning... It's much harder to do this when you must specify the groups via a list: `[1]`, `[0,1,2,3]`, etc. `[-1, -2]` would just be invalid groups. `[]` is no groups. How do you specify no groups in the current scheme of using negative numbers?

Reply:
I thought about using list as well, but it looks not so lightweight in the request json:
```
{
  chainId: "alphieum:1/-1" // instead of "alephium:1/[0,1,2,3]"
}
```

Reply:
> The only issue with `-1` at the protocol level is a typo: if someone meant `1` but typos `-1` (for whatever reason, maybe they are tired), it will have very, very different meaning... It's much harder (impossible) to do this when you must specify the groups via a list: `[1]`, `[0,1,2,3]`, etc. `[-1, -2]` would just be invalid groups. `[]` is no groups. How do you specify no groups in the current scheme of using negative numbers?

By protocol level, I meant the message protocol designed by Wallectconnect.

I agree that devs should never have to write `-1`, but we will encode it as `-1`. For example:
```
export function formatChain(networkId: number, chainGroup?: number): string {
  return `${WalletConnectProvider.namespace}:${networkId}/${chainGroup ? chainGroup : -1}`;
}
```

Reply:
Yeah I know - as I said, at the moment before transmission the conversion can happen

Reply:
Then it's about the signature of the function `formatChain`:
1. export function formatChain(networkId: number, chainGroup?: number): string 
2. export function formatChain(networkId: number, chainGroups: number[]): string
Option 1 is much simpler for now.

Reply:
I would rather do it properly than what's simpler

/cc @mvaivre @nop33 may have different opinions (it'll change my view on if things should be done properly or quickly for this project)

Reply:
The simple function covers these two cases properly:
1. single group
2. all groups
The other cases should be discouraged IMO.

Reply:
Ah ok, this changes things then... Let's go for the 1st then as it properly narrows the cases to the ones you mentioned :slightly_smiling_face: